catroid/src/main/java/org/catrobat/catroid/content/bricks/UserDefinedBrick.java

```diff
@@ -55,144 +55,176 @@
+			if (data instanceof UserDefinedBrickInput) {
+				userDefinedBrickDataList.add(new UserDefinedBrickInput((UserDefinedBrickInput) data));
+			} else {
+				userDefinedBrickDataList.add(new UserDefinedBrickLabel((UserDefinedBrickLabel) data));
+			}
+		}
 	}
 
-	public UserDefinedBrick(List<UserBrickData> userBrickDataList) {
-		this.userBrickDataList = userBrickDataList;
+	public UUID getUserDefinedBrickID() {
+		return userDefinedBrickID;
 	}
 
-	public void addLabel(Nameable label) {
+	public void addLabel(String label) {
 		removeLastLabel();
-		userBrickDataList.add(new UserBrickLabel(label));
+		userDefinedBrickDataList.add(new UserDefinedBrickLabel(label));
 	}
 
 	public void removeLastLabel() {
 		if (lastContentIsLabel()) {
-			userBrickDataList.remove(userBrickDataList.size() - 1);
+			userDefinedBrickDataList.remove(userDefinedBrickDataList.size() - 1);
 		}
 	}
 
-	public void addInput(Nameable input) {
-		userBrickDataList.add(new UserBrickInput(input));
+	public void addInput(String input) {
+		userDefinedBrickDataList.add(new UserDefinedBrickInput(input));
 	}
 
 	public boolean isEmpty() {
-		return userBrickDataList.isEmpty();
+		return userDefinedBrickDataList.isEmpty();
 	}
 
-	public List<Nameable> getUserDataList(boolean inputOrLabel) {
-		List<Nameable> userDataList = new ArrayList<>();
+	public List<UserDefinedBrickData> getUserDefinedBrickDataList() {
+		return this.userDefinedBrickDataList;
+	}
+
+	public List<String> getUserDataList(boolean inputOrLabel) {
+		List<String> userDataList = new ArrayList<>();
 		if (inputOrLabel) {
-			for (UserBrickData userBrickData : userBrickDataList) {
-				if (userBrickData instanceof UserBrickInput) {
-					userDataList.add(((UserBrickInput) userBrickData).getInput());
+			for (UserDefinedBrickData userDefinedBrickData : userDefinedBrickDataList) {
+				if (userDefinedBrickData instanceof UserDefinedBrickInput) {
+					userDataList.add(((UserDefinedBrickInput) userDefinedBrickData).getInput());
 				}
 			}
 		} else {
-			for (UserBrickData userBrickData : userBrickDataList) {
-				if (userBrickData instanceof UserBrickLabel) {
-					userDataList.add(((UserBrickLabel) userBrickData).getLabel());
+			for (UserDefinedBrickData userDefinedBrickData : userDefinedBrickDataList) {
+				if (userDefinedBrickData instanceof UserDefinedBrickLabel) {
+					userDataList.add(((UserDefinedBrickLabel) userDefinedBrickData).getLabel());
 				}
 			}
 		}
 		return userDataList;
 	}
 
 	private boolean lastContentIsLabel() {
-		if (userBrickDataList.isEmpty()) {
+		if (userDefinedBrickDataList.isEmpty()) {
 			return false;
 		}
-		return userBrickDataList.get(userBrickDataList.size() - 1) instanceof UserBrickLabel;
+		return userDefinedBrickDataList.get(userDefinedBrickDataList.size() - 1) instanceof UserDefinedBrickLabel;
+	}
+
+	public boolean isUserDefinedBrickDataEqual(UserDefinedBrick other) {
+		if (userDefinedBrickDataList.size() == other.userDefinedBrickDataList.size()) {
```

Comment:
Please rewrite this in a more clean code style. Early returning, dont use too much nested blocks